mantidproject/mantid
# Fractional reference bin indices for correcting TOF axis

**Description of work.**

This PR implements a modification of the direct geometry workflow to improve the precision of the elastic peak calibration. Previously, only the elastic peak time channel was communicated with the CorrectTOFAxis algorithm, which limits the precision of the calibration to the time channel with. In result, the elastic peak centres were observed to not fall at 0 energy exchange but around 0.1 meV away from it.

A solution proposed in this PR adds a possiblity to specify a fractional offset that allows to profit from the full information about the peak centre position obtained from elastic peak fitting. The result was initially validated with Stephane Rols and Bjorn Fak, and the proposed solution was found superior to the one currently available in the code.

**To test:**



1. Code review, tests should be passing.
2. Get the attached file and run the following script in the workbench:

```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import directtools as dt

elp = CreateSingleValuedWorkspace(DataValue=103.42)

DirectILLCollectData( 
        Run = '026079',
        IncidentEnergyCalibration='Energy Calibration ON',
        ElasticChannelWorkspace=elp,  
        OutputWorkspace = 'test'
)
DirectILLReduction(
            InputWorkspace='test',
            OutputWorkspace="out"
)
Q= 2.85
dQ= 0.35 # this means Q in [2.5,3.2]
plot= "out"
fig, axes, cuts = dt.plotconstQ(plot,Q,dQ)
```
The maximum of the plotted distribution should be now very close to 0.0, while when the data value of the ELP is changed to the nearest integer, the peak would fall at about 0.1 meV.

Fixes #33222.
[026079.nxs.gz](https://github.com/mantidproject/mantid/files/7830083/026079.nxs.gz)

Release note added.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
